tests/integration/targets/aci_domain_to_encap_pool/tasks/main.yml

```diff
@@ -0,0 +1,577 @@
+    - l3dom is changed
+    - vmm_cloudfoundry is changed
+    - vmm_kubernetes is changed
+    - vmm_openshift is changed
+    - vmm_openstack is changed
+    - vmm_redhat is changed
+    - vmm_vmware is changed
+    - vmm_vmware_idemp is not changed
+    - phys_vlan.current.0.physDomP.children.0.infraRsVlanNs.attributes.tDn =='uni/infra/vlanns-[test_pool]-dynamic'
+    - fc_vsan.current.0.fcDomP.children.0.fcRsVsanNs.attributes.tDn == 'uni/infra/vsanns-[test_pool]-dynamic'
+    - l2dom.current.0.l2extDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-static'
+    - l3dom.current.0.l3extDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-static'
+    - vmm_cloudfoundry.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - vmm_kubernetes.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - vmm_microsoft.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - vmm_openshift.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - vmm_openstack.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - vmm_redhat.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - vmm_vmware.current.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - phys_vlan_with_vmm.msg is match("Domain type '[0-9a-zA-Z]*' cannot have a 'vm_provider'")
+    - vmm_vmware_vxlan.msg is match("vxlan pools do not support setting the allocation_mode; please remove this parameter from the task")
+    - phys_vlan_no_alloc.msg is match("ACI requires the 'pool_allocation_mode' for 'pool_type' of 'vlan' and 'vsan' when 'pool' is provided")
+
+# QUERY
+- name: Query phys
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: phys
+    pool_type: vlan
+    state: query
+  register: query_phys
+
+- name: Query fc
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: fc
+    pool_type: vsan
+    state: query
+  register: query_fc
+
+- name: Query l2dom
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: l2dom
+    pool_type: vlan
+    state: query
+  register: query_l2dom
+
+- name: Query l3dom
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: l3dom
+    pool_type: vlan
+    state: query
+  register: query_l3dom
+
+- name: Query vmm
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: vmm
+    pool_type: vlan
+    vm_provider: microsoft
+    state: query
+  register: query_vmm
+
+- name: Verify Query
+  assert:
+    that:
+    - query_phys.current | length >= 1
+    - query_fc.current | length >= 1
+    - query_l2dom.current | length >= 1
+    - query_l3dom.current | length >= 1
+    - query_vmm.current | length >= 7
+
```

Comment:
we need to check query with a specific domain_name and also with a name that does not exist to see the error.